spec/support/logstash_test.rb

```diff
@@ -21,3 +21,13 @@ def random_port
     end
   end
 end
+
+def testing_thread(&block)
+  th = Thread.new do
+    block.call
+  end
+  #
+  # wait for the thread to spin
+  sleep(0.1)
```

Comment:
If you explicitly want to yield, it's probably best to use a ConditionVariable here and have the thread wake you up when it's started.